automation/scripts/github_branch_autosync/github_autosync/gcloud_entrypoint/main.py

```diff
@@ -0,0 +1,80 @@
+''' Main module for handling incoming github webhook event'''
+
+from .lib import GithubPayloadInfo, config, GithubApi, GithubException, verify_signature,is_push_event
+
+def handle_incoming_commit_push(request):
+    """Responds to any HTTP request.
+    Args:
+        request (flask.Request): HTTP request object.
+    Returns:
+        The response text or any set of values that can be turned into a
+        Response object using
+        `make_response <http://flask.pocoo.org/docs/1.0/api/#flask.Flask.make_response>`.
+    """
+    verify_signature(request.data, config.github["secret"], request.headers['x-hub-signature-256'])
+    if not is_push_event(request):
+        print("not a push event. skipping...")
+        return
+
+    handle_incoming_commit_push_json(request.json,config=config)
+    print("done")
+    return
+
+def handle_incoming_commit_push_json(json,config):
+    """
+      Main logic for handling incoming github webhook event
+    """
+    payload_info=  GithubPayloadInfo(json)
+
+    source_branch = payload_info.incoming_branch
+
+    if not source_branch in config.branches:
+        print(f"change in '{source_branch}' is not supported ")
+        return
+
+    target_branch = config.branches[source_branch]
+    github = GithubApi(config.github)
+    print(f"generating diff between {source_branch} and '{target_branch}'...")
+    cmp = github.get_diff_commits(target_branch,source_branch)
+
+    if cmp.status == "identical":
```

Comment:
I think Python has a `match ... case` statement that might be useful here

Reply:
ok, i will fix that in next pr not to block dryrun testing